test/hotspot/jtreg/containers/cgroup/NestedCgroup.java

```diff
@@ -0,0 +1,268 @@
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import jdk.internal.platform.Metrics;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.JDKToolFinder;
+import jdk.test.lib.Platform;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jtreg.SkippedException;
+
+public class NestedCgroup {
+    private static abstract class Test {
+        public static final String CGROUP_OUTER = "jdktest" + ProcessHandle.current().pid();
+        public static final String CGROUP_INNER = "inner";
+        public static final String CONTROLLERS_PATH_OUTER = "memory:" + CGROUP_OUTER;
+        public static final String CONTROLLERS_PATH_INNER = CONTROLLERS_PATH_OUTER + "/" + CGROUP_INNER;
+        public static final String LINE_DELIM = "-".repeat(80);
+        public static final String MOUNTINFO = "/proc/self/mountinfo";
+
+        // A real usage on x86_64 fits in 39 MiB.
+        public static final int MEMORY_MAX_OUTER = 500 * 1024 * 1024;
+        public static final int MEMORY_MAX_INNER = MEMORY_MAX_OUTER * 2;
+
+        class Limits {
+            int integer;
+        };
+
+        public static String sysFsCgroup;
+        public String memory_max_filename;
+        public static boolean isCgroup2;
+
+        public static void lineDelim(String str, String label) {
+            System.err.print(LINE_DELIM + " " + label + "\n" + str);
+            if (!str.isEmpty() && !str.endsWith("\n")) {
+                System.err.println();
+            }
+        }
+
+        public static OutputAnalyzer pSystem(List<String> args,
+                String failStderr, String failExplanation,
+                String fail2Stderr, String fail2Explanation,
+                String ignoreStderr) throws Exception {
+            System.err.println(LINE_DELIM + " command: " + String.join(" ",args));
+            ProcessBuilder pb = new ProcessBuilder(args);
+            Process process = pb.start();
+            OutputAnalyzer output = new OutputAnalyzer(process);
+            int exitValue = process.waitFor();
+            lineDelim(output.getStdout(), "stdout");
+            lineDelim(output.getStderr(), "stderr");
+            System.err.println(LINE_DELIM);
+            if (!failStderr.isEmpty() && output.getStderr().equals(failStderr + "\n")) {
+                throw new SkippedException(failExplanation + ": " + failStderr);
+            }
+            if (!fail2Stderr.isEmpty() && output.getStderr().equals(fail2Stderr + "\n")) {
+                throw new SkippedException(fail2Explanation + ": " + fail2Stderr);
+            }
+            if (!ignoreStderr.isEmpty() && output.getStderr().equals(ignoreStderr + "\n")) {
+                return output;
+            }
+            Asserts.assertEQ(0, exitValue, "Process returned unexpected exit code: " + exitValue);
+            return output;
+        }
```

Comment:
This could be simplified to just `ProcessTools.executeProcess(new ProcessBuilder(args));`